src/Compilers/CSharp/Portable/Binder/UsingStatementBinder.cs

```diff
@@ -76,13 +75,34 @@ internal override BoundStatement BindUsingStatementParts(DiagnosticBag diagnosti
                 expressionOpt = this.BindTargetExpression(diagnostics, originalBinder);
 
                 HashSet<DiagnosticInfo> useSiteDiagnostics = null;
+                TypeSymbol expressionType = expressionOpt.Type;
+                MethodSymbol disposeMethod = expressionType == null ? null : TryFindDisposePattern(expressionOpt.Type, diagnostics);
```

Comment:
I would call this `TryGetDisposePatternMethod` since we're actually trying to get a specific method, not a pattern.